aggregator_core/src/datastore/tests.rs

```diff
@@ -5176,6 +5176,9 @@ async fn roundtrip_outstanding_batch(ephemeral_datastore: EphemeralDatastore) {
                         .unwrap(),
                     BatchAggregationState::Aggregating {
                         aggregate_share: Some(dummy::AggregateShare(0)),
+                        // Let report_count be 1 without an accompanying report_aggregation in a
+                        // terminal state. This captures the case where a FINISHED report_aggregation
+                        // was garbage collected and no longer exists in the database.
                         report_count: 1,
```

Comment:
I believe this test was actually somewhat reproducing the bug, except our assertion was wrong.

This batch doesn't have an accompanying report_aggregation in a FINISHED state to go along with the `report_count: 1`. We only have one in a START state. So in the test we were checking that the range was `[0, 1]` where it should have been `[1, 2]`.